ledger/error/src/main/scala/com/daml/error/ErrorCode.scala

```diff
@@ -50,6 +51,23 @@ abstract class ErrorCode(val id: String, val category: ErrorCategory)(implicit
 
   implicit val code: ErrorCode = this
 
+  /** @return whether the supplied exception matches this error code.
+    */
+  def matches(e: StatusRuntimeException): Boolean = {
```

Comment:
Yes, for INTERNALs we deliberately strip all the information that would enable us to track down the corresponding error code

Reply:
This might be confusing to users of this API if they're not intimate with the way error codes work. Also, this function would better fit in a client-side library since it should be only used in testing or in client applications.

Reply:
I could move it somewhere into `//ledger/error/src/test/lib` but then it would have to be a harder to find and less pretty two argument function.

Reply:
Related to the location of this utility function, the core error codes API (in `//ledger/error`) has two functions (intertwined unfortunately, but that's another story): 
1. Define the interfaces of error code DTOs
2. Constructs gRPC-domain objects to be sent to the client
The function added in this PR is about destructuring a gRPC-domain object which is the opposite of 2, so I would see it not well-placed here.

I would prefer this integrated in some other target/utility object at least that is aimed for clients. However, I wouldn't block this PR on this if you consider otherwise. 

In any case, please update the comment and/or method signature to highlight its restriction to non-security-sensitive errors.

Reply:
Added comment explaining the limitations